README.md

```diff
@@ -100,11 +100,62 @@ This project is equipped with a Continuous Integration (CI) pipeline using GitHu
 -   Test: Tests the application to ensure that new changes do not introduce regressions.
 -   Linting: The code is analyzed for adherence to coding standards and best practices, using [eslint](https://eslint.org/), [markdownlint](https://github.com/DavidAnson/markdownlint), and [commitlint](https://github.com/conventional-changelog/commitlint).
 
+See [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) for a more detailed explaination.
+
 When opening a PR, be mindful of (and make necessary changes in accordance with) these checks.
 
 ### Releases
 
-Releases are managed via CI and [semantic release](https://github.com/semantic-release/semantic-release).
+Releases are managed via CI and [Changesets](https://github.com/changesets/changesets). 
+
+When contributing to the project, developers should add a changeset.
+
+A changeset is a Markdown file with YAML front matter. The contents of the Markdown is the change summary which will be written to the changelog and the YAML front matter describes what packages have changed and what SemVer bump types they should be
+
+```md
+---
+"@d3fc": major
+"@d3fc/d3fc-annotation": minor
+---
+
+Change all the things
+```
+
+Changesets should include: 
+
+* WHAT the change is
+* WHY the change was made
+* HOW a consumer should update their code 
+
+#### Adding a changeset
+To add a changeset, before putting up a PR, contributors should: 
```

Comment:
Not really a comment, just been looking at how this works and want to check I understand something. So the changeset _doesn't_ need to be done just before merging to capture the most recent commit or anything?

My understanding is that the changeset will still capture all the commits on that PR, even if some commits were done after the changeset was created - and as long as the PR doesn't change drastically to the point where the changeset _description_ is no longer accurate then it should all work?

Suppose we could also add a github action check for this in future to check it's been done right.

Reply:
I've just done a test - the `changeset` script creates a changeset markdown file, in the format of

```md
---
'd3fc': minor
---

Test feature
``` 

`changeset publish` then captures all the commits - so as far as I am aware, you are correct and as long as commits after the changeset file is created don't touch another package or change anything drastically, contributors won't need to change the changeset. 

There's a changeset bot and github action that we can use for checking changesets - #1866

Reply:
Ok thanks, based on what it generates that seems to be the way it works, cool.